tests/observation/test_grdc.py

```diff
@@ -78,9 +81,54 @@ def test_get_grdc_data(tmp_path, sample_grdc_file):
             'river_name': 'SOME RIVER',
             'station_name': 'SOME',
             'time_series': '2000-01 - 2000-01',
-            'units': 'm�/s'
+            'units': 'm³/s',
+            'UserEndTime': '2000-02-01T00:00Z',
+            'UserStartTime': '2000-01-01T00:00Z',
+            'nrMissingData': 1,
         }
-    )
+    return data, metadata
+
+
+def test_get_grdc_data_with_datahome(tmp_path, expected_results):
+    expected_data, expected_metadata = expected_results
+    result_data, result_metadata = get_grdc_data('42424242', '2000-01-01T00:00Z', '2000-02-01T00:00Z', data_home=tmp_path)
+
+    assert_frame_equal(result_data, expected_data)
+    assert result_metadata == expected_metadata
+
+
+def test_get_grdc_data_with_CFG(expected_results, tmp_path):
+    CFG['grdc_location'] = str(tmp_path)
+    expected_data, expected_metadata = expected_results
+    result_data, result_metadata = get_grdc_data('42424242', '2000-01-01T00:00Z', '2000-02-01T00:00Z')
+
+    assert_frame_equal(result_data, expected_data)
+    assert result_metadata == expected_metadata
+
+
+def test_get_grdc_data_without_path():
+    CFG['grdc_location'] = None
+    with pytest.raises(ValueError) as excinfo:
+        get_grdc_data('42424242', '2000-01-01T00:00Z', '2000-02-01T00:00Z')
+    msg = str(excinfo.value)
+    print(msg)
+    assert 'data_home' in msg
+    assert 'grdc_location' in msg
+
+
+def test_get_grdc_data_wrong_path(tmp_path):
+    CFG['grdc_location'] = f'{tmp_path}_data'
+
+    with pytest.raises(ValueError) as excinfo:
+        get_grdc_data('42424242', '2000-01-01T00:00Z', '2000-02-01T00:00Z')
+    msg = str(excinfo.value)
+    print(msg)
+    assert 'directory' in msg
+
 
-    assert_equal(result, expected)
-    assert result.attrs == expected.attrs
+def test_get_grdc_data_without_file(tmp_path):
+    with pytest.raises(ValueError) as excinfo:
+        get_grdc_data('42424243', '2000-01-01T00:00Z', '2000-02-01T00:00Z', data_home=tmp_path)
```

Comment:
tmp_path is Path object and data_home should be str. Either stringify or allow Path in data_home typing
```suggestion
        get_grdc_data('42424243', '2000-01-01T00:00Z', '2000-02-01T00:00Z', data_home=str(tmp_path))
```